conf/app.routes

```diff
@@ -19,6 +19,6 @@ GET        /movements/departures/:departureId                       controllers.
 POST       /movements/departures/:departureId/messages              controllers.DepartureMessagesController.sendMessageDownstream(departureId: DepartureId)
 GET        /movements/departures/:departureId/messages              controllers.DepartureMessagesController.getDepartureMessages(departureId: DepartureId, receivedSince: Option[OffsetDateTime] ?= None)
 
-GET        /movements/departures/:departureId/messages/:messageId   controllers.DepartureMessagesController.getDepartureMessage(departureId: DepartureId, messageId: MessageId)
+GET        /movements/departures/:departureId/messages/:messageId   routing.DeparturesRouter.getMessage(departureId: String, messageId: String)
```

Comment:
Just wondering why you chose to declare departureId/messageId as strings rather than case classes extending anyval?

Reply:
See my other comment - basically because I do the binding myself once I've decided which version to send it on to.

Reply:
To me that suggests we could probably do a refactor to make the Id-types work on the same basis across P4 and P5. Means we can have a common pattern now and avoid the tidy-up before handover to live when P5 drops

Reply:
> To me that suggests we could probably do a refactor to make the Id-types work on the same basis across P4 and P5.

I'm not sure how you would want to refactor this.

* If you wanted to unify each of the Id types so it was common across V1 and V2, I'd be against that because they are fundamentally different types, and it means adding more code elsewhere to ensure correctness later.
* If you wanted to add a way to unify this handling, that I can see an argument of, but I'm wary about potentially wading into "clever code" for the sake of it.
* Is there something else you had in mind?

While we have come up with some clever tricks in this codebase, I'm not really sure we need to do much more here. Once P5 is live and P4 is dead to us, I'd imagine this router will be deleted, along with the P4 code, and the routes file will be altered to just support the V2 routes (along with the reverse routing breaks that will occur in the tests and all...). By doing so, this would just outright delete the binding selection and we can just use the right model in this routes file. I'd be happy to include the P5 only route as a comment.